src/expr/node.cpp

```diff
@@ -137,9 +137,72 @@ bool NodeTemplate<ref_count>::hasBoundVar() {
   return getAttribute(HasBoundVarAttr());
 }
 
+template <bool ref_count>
+bool NodeTemplate<ref_count>::hasFreeVar()
```

Comment:
Add `const` since it does not change the Node itself.

Reply:
Actually, since this function calls hasBoundVar(), it can't be const.  The reason is we cache an attribute on the Node that doesn't allow "const".

Reply:
Makes sense.